### DISCUSSION TITLE
how to seperate comment inputs

### DISCUSSION THREAD

**Discussion by janneserichsen** [neuefische/questions#476] [Web]
Hello Guys i want to be able to comment under my Posts this works but when i post under one Picture the comment stands under every other Picture and i dont really know how to fix this Problem can someone help me

<img width="409" alt="Bildschirmfoto 2023-01-31 um 14 36 17" src="https://user-images.githubusercontent.com/118190990/215775141-ffebbdc6-ba53-4ed4-b5d2-42929a50a998.png">
<img width="506" alt="Bildschirmfoto 2023-01-31 um 14 36 10" src="https://user-images.githubusercontent.com/118190990/215775159-82460a75-640b-4eb1-9e1f-cfd76d3d0dc0.png">
<img width="597" alt="Bildschirmfoto 2023-01-31 um 14 36 00" src="https://user-images.githubusercontent.com/118190990/215775177-ea9dd189-577e-4f91-a475-fa5fc664927b.png">

**Comment by NikRoe** [comment]
Hi Jannes,

you currently have one state that gets used for every graffitiPost and your comments are not assigned to any graffiti post.

You have two options:

1. You can keep that one state but need to add comments that can be identified by the graffiti post they were posted to.
Your comments will most likely be objects that looks similar to this: 
```js
{comment: "super style", graffitiPostId: "98d9sdsksd"}
```
You would then need to filter which comment belongs to which graffitiPost and then pass that filtered amount of data as a prop to your list to render them properly.

2. You can create a component (something like GraffitiCard) that renders your List and Form component and has its own state. Your comments could maintain the data type they currently have and would not need to be objects for this to work properly.
In each component you would handle the adding of a comment for the respective GraffitiCard and those could be easily rendered by your List component, since they are part of the same parent component where the state is located.

It would make sense to create a component out of the whole section you currently return:

```js
export default function GraffitiCard() {
           <section key={graffiti.id}>
            <h2>Post by {currentUser.name}</h2>
            <p>Location: {graffiti.location}</p>
            <Image
              src={graffiti.image}
              alt={`Graffiti from ${currentUser.name}`}
              width={250}
              height={250}
            />
            <p>{graffiti.comment}</p>
           <List tags={tags} />
           <Form onSubmit={handleAddTag} />
          </section>
}
```

This component would only need its own state and in your `/pages/index.js` you would need to map over every graffitiPost to render your new component for each existing post.

Hope this helps! :)
